### PR TITLE
Bump client-base to v0.4.0

### PR DESCRIPTION
**Pull request by llucax**
We need this to be able to use it with `grpclib` instead of `grpcio`.

### CODE REVIEW

**Comment by llucax** [comment]
Enabling auto-merge.

File: src/frequenz/client/microgrid/_client.py [hunk 1]
```diff
@@ -261,20 +261,21 @@ async def _new_component_data_receiver(
...
+        broadcaster = self._broadcasters.get(component_id)
+        if broadcaster is None:
+            broadcaster = streaming.GrpcStreamBroadcaster(
                 f"raw-component-data-{component_id}",
                 # We need to cast here because grpc says StreamComponentData is
                 # a grpc.CallIterator[PbComponentData], not a
                 # grpc.aio.UnaryStreamCall[..., PbComponentData].
                 lambda: cast(
-                    grpc.aio.UnaryStreamCall[Any, PbComponentData],
+                    AsyncIterator[PbComponentData],
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/client/microgrid/_client.py:273]: The comments above seems outdated with the current changes

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I have a minor comment to check for. LGTM otherwise

File: src/frequenz/client/microgrid/_client.py (hunk 1 above)
Comment (llucax) [reply src/frequenz/client/microgrid/_client.py:273]: Ah, yeah, thanks! Will update!

**Comment by llucax** [comment]
Updated.

File: src/frequenz/client/microgrid/_client.py [hunk 2]
```diff
@@ -266,11 +266,11 @@ async def _new_component_data_receiver(
...
-                # a grpc.CallIterator[PbComponentData], not a
-                # grpc.aio.UnaryStreamCall[..., PbComponentData].
-                lambda: cast(
-                    grpc.aio.UnaryStreamCall[Any, PbComponentData],
-                    self.api.StreamComponentData(PbComponentIdParam(id=component_id)),
+                # a grpc.CallIterator[PbComponentData] which is not an AsyncIterator,
+                # but it is a grpc.aio.UnaryStreamCall[..., PbComponentData], which it
+                # is.
+                lambda: self.api.StreamComponentData(
+                    PbComponentIdParam(id=component_id)
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/client/microgrid/_client.py]: The tests/checks failed, maybe the cast is missing?

From the tests/checks logs: `src/frequenz/client/microgrid/_client.py:8:1: F401 'collections.abc.AsyncIterator' imported but unused`

Comment (llucax) [reply src/frequenz/client/microgrid/_client.py]: Yeah, sorry, I just realized that I was trying out if the cast wasn't necessary anymore, but it is, I just updated again, hopefully this time is the last time :D

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**